### PR TITLE
feat: Stream numeric base overloads — print(val, DEC/HEX/OCT/BIN)

### LINKED ISSUE #102: Stream missing numeric base overloads: `print(unsigned long, int)` etc.
## Problem

The real Arduino `Print` class provides numeric print overloads that accept a base argument (`DEC`, `HEX`, `OCT`, `BIN`):

```cpp
size_t print(unsigned long, int = DEC);
size_t print(long, int = DEC);
size_t print(int, int = DEC);
size_t print(unsigned int, int = DEC);
```

ArduinoMock's `Stream` only has:

```cpp
virtual size_t print(int value) { return print(String(value)); }
virtual size_t print(uint16_t value) { return print(String(value)); }
```

This means any native build that calls e.g. `Serial.print(ESP.getFreeHeap(), DEC)` or `Serial.print(someUnsignedLong, HEX)` fails:

```
error: no matching function for call to 'HardwareSerial::print(size_t, int)'
```

## Expected behaviour

`Stream` (and by extension `HardwareSerial`) should support the standard two-argument numeric print overloads.

## Suggested fix

Add to `Stream.h`:

```cpp
size_t print(unsigned long val, int base) {
  char buf[64];
  int n = (base == 16) ? snprintf(buf, sizeof(buf), "%lX", val)
        : (base ==  8) ? snprintf(buf, sizeof(buf), "%lo", val)
        : snprintf(buf, sizeof(buf), "%lu", val);
  return write(reinterpret_cast<const uint8_t*>(buf), n > 0 ? (size_t)n : 0);
}
size_t print(long val, int base) {
  if (val < 0 && base == 10) { size_t n = print("-"); return n + print((unsigned long)(-val), base); }
  return print((unsigned long)val, base);
}
size_t print(int val, int base)          { return print((long)val, base); }
size_t print(unsigned int val, int base) { return print((unsigned long)val, base); }
```

## Context

- ArduinoMock version: `rc-0.2.0` (sha `620a391`)
- Discovered while building an ESP32 firmware in a PlatformIO `platform = native` simulator environment.

### PR DESCRIPTION
**Pull request by Nathan-ma**
## Summary

Closes #102.

- Adds four `print(val, base)` overloads to `Stream` (inline in `Stream.h`):
  - `print(unsigned long val, int base)` — formats via `snprintf` with `%lX` / `%lo` / `%lu`
  - `print(long val, int base)` — handles negative values in base 10 by prepending `"-"`
  - `print(int val, int base)` — delegates via `long`
  - `print(unsigned int val, int base)` — delegates via `unsigned long`
- All four overloads support `DEC` (10), `HEX` (16), `OCT` (8), and `BIN` (2) constants already defined in `WString.h`.

## Test plan

Five new tests added to `test/hardware_serial_gtest.cpp`:

- [x] `PrintUnsignedDecimal` — `print(42u, DEC)` → `"42"`
- [x] `PrintUnsignedHex` — `print(255u, HEX)` → `"FF"`
- [x] `PrintUnsignedLongOctal` — `print(8ul, OCT)` → `"10"`
- [x] `PrintNegativeSignedDecimal` — `print(-1, DEC)` → `"-1"`
- [x] `PrintSignedHex` — `print(255, HEX)` → `"FF"` (exercises the `long` → `unsigned long` path)

All 23 tests in `hardware_serial_gtest` pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/Stream.h [hunk 1]
```diff
@@ -23,6 +24,23 @@ class Stream {
...
+    char buf[64];
+    int n = (base == 16)  ? snprintf(buf, sizeof(buf), "%lX", val)
+            : (base == 8) ? snprintf(buf, sizeof(buf), "%lo", val)
+                          : snprintf(buf, sizeof(buf), "%lu", val);
+    return write(reinterpret_cast<const uint8_t*>(buf), n > 0 ? static_cast<size_t>(n) : 0);
+  }
+  size_t print(long val, int base) {
+    if (val < 0 && base == 10) {
+      size_t n = print("-");
+      return n + print(static_cast<unsigned long>(-val), base);
```
[the remark this answers is not in the document]
Comment (Nathan-ma) [reply src/Stream.h]: Fixed — replaced `static_cast<unsigned long>(-val)` with the two-step `static_cast<unsigned long>(-(val + 1)) + 1UL` trick, which is well-defined even for `LONG_MIN`. Added a `PrintLongMinDecimal` test to cover this edge case.

File: src/Stream.h [hunk 2]
```diff
@@ -23,6 +24,23 @@ class Stream {
   virtual size_t print(int value) { return print(String(value)); }
   virtual size_t print(uint16_t value) { return print(String(value)); }
 
+  size_t print(unsigned long val, int base) {
+    char buf[64];
+    int n = (base == 16)  ? snprintf(buf, sizeof(buf), "%lX", val)
+            : (base == 8) ? snprintf(buf, sizeof(buf), "%lo", val)
+                          : snprintf(buf, sizeof(buf), "%lu", val);
```
[the remark this answers is not in the document]
Comment (Nathan-ma) [reply src/Stream.h]: Fixed — added an explicit `base == 2` path with a manual bit-reversal loop (accumulate bits LSB-first into a temp buffer, then reverse into `buf`). `print(5u, BIN)` now produces `"101"` and `print(0u, BIN)` produces `"0"`.

File: test/hardware_serial_gtest.cpp [hunk 3]
```diff
@@ -145,6 +145,38 @@ TEST(HardwareSerialTest, InstancesAreIndependent) {
...
+  s.print(-1, DEC);
+  EXPECT_EQ(s.getTxData(), "-1");
+}
+
+TEST(HardwareSerialTest, PrintSignedHex) {
+  HardwareSerial s(0);
+  s.print(255, HEX);
+  EXPECT_EQ(s.getTxData(), "FF");
+}
+
```
[the remark this answers is not in the document]
Comment (Nathan-ma) [reply test/hardware_serial_gtest.cpp:181]: Fixed — added `PrintUnsignedBinary` (`5u → "101"`) and `PrintZeroBinary` (`0u → "0"`) tests. Also added `PrintLongMinDecimal` to cover the UB fix.
